### PR TITLE
When limit is set but request is not treat them as matching

### PR DESCRIPTION
**Pull request by jbartosik**
@kgolab pointed out that when container has limit and no request it's treated as if it had request equal to limit.

### CODE REVIEW

File: vertical-pod-autoscaler/pkg/admission-controller/logic/recommendation_provider.go [hunk 1]
```diff
@@ -60,6 +61,19 @@ func NewRecommendationProvider(vpaLister vpa_lister.VerticalPodAutoscalerLister,
 	}
 }
 
+func limitMatchesRequest(limit, request *resource.Quantity) bool {
+	// Limit not set, it cannot match the request.
+	if limit == nil || limit.Value() == 0 {
+		return false
+	}
+	// Limit set but request not set - K8s interprets will treat the pod as if it had them equal.
```
Comment (kgolab) [inline vertical-pod-autoscaler/pkg/admission-controller/logic/recommendation_provider.go]: nit: grammar seems broken in "K8s interprets..."

Comment (jbartosik) [reply vertical-pod-autoscaler/pkg/admission-controller/logic/recommendation_provider.go]: Fixed.

**Comment by kgolab** [comment]
LGTM, thanks.

**Comment by bskiba** [comment]
So the pod with empty request and limit set is treated as guaranteed?

**Comment by jbartosik** [comment]
> So the pod with empty request and limit set is treated as guaranteed?

Yes: https://kubernetes.io/docs/tasks/administer-cluster/manage-resources/memory-default-namespace/#what-if-you-specify-a-container-s-limit-but-not-its-request

**Comment by bskiba** [comment]
Actually, the documentation you point to states that the pod will be modified to have request set, so AFAIU we should not actually in the admission controller see the case when request = 0 and limit > 0

**Comment by jbartosik** [comment]
> Actually, the documentation you point to states that the pod will be modified to have request set, so AFAIU we should not actually in the admission controller see the case when request = 0 and limit > 0

In a 1.12 cluster I created a pod specifying limit but no request. Admission controller received a pod with limit = request.

**Pull request merged**